src/EditorFeatures/Core/Implementation/PasteTracking/PasteTrackingPasteCommandHandlercs.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.PasteTracking
+{
+    [Export]
+    [Export(typeof(VSCommanding.ICommandHandler))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [Name(PredefinedCommandHandlerNames.PasteTrackingPaste)]
+    [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
+    [Order(Before = PredefinedCommandHandlerNames.Completion)]
+    internal class PasteTrackingPasteCommandHandler : IChainedCommandHandler<PasteCommandArgs>
+    {
+        public string DisplayName => EditorFeaturesResources.Paste_Tracking;
+
+        private readonly PasteTrackingService _pasteTrackingService;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        internal PasteTrackingPasteCommandHandler(IPasteTrackingService pasteTrackingService)
+        {
+            _pasteTrackingService = (PasteTrackingService)pasteTrackingService;
+        }
```

Comment:
surprising.  Why do you need to get the underlying type here.  Feels leaky.  
If you do really need hte underlying type, you could just export that type and have the contructor import the tyep directly.